### DISCUSSION TITLE
Unix Domain Socket

### DISCUSSION THREAD

**Discussion by jvary** [rabbitmq/rabbitmq-server#7298] [Ideas]
{repost from rabbitmq-users mailing list}

Hello all!

I am looking to add support for Unix Domain Socket in RabbitMQ, to add extra safety & segregation when used locally (versus listening on 127.0.0.1).
In a 2018 message on the mailing list, it was suggested to do an “exchange plugin“, but since then, ranch_tcp added support for inet:local_address().

My plan would be to change the listeners configuration ‘pipe’ to support inet:local_address() up to ranch, and modify the Erlang & C# client libs to support it as well.

Before I jump into the code, any chance that such PRs could reach the main branch ?

Julien

**Comment by lukebakken** [comment]
Please see #7311. Have fun coding 😸
